File: php/authentication.php
<?php

$url = 'https://api.smartproxy.com/v1/auth';

// Your smartproxy credentials

$auth = base64_encode("username:password");

$ch = curl_init();

curl_setopt($ch, CURLOPT_URL, $url);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
curl_setopt($ch, CURLOPT_CUSTOMREQUEST, "POST");

$headers = array();
$headers[] = "Accept: application/json";
$headers[] = "Authorization: Basic $auth";
curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);

$result = curl_exec($ch);
if (curl_errno($ch)) {
    echo 'Error:' . curl_error($ch);
}
curl_close($ch);
echo $result;

File: csharp/authentication.cs
using System;
using System.IO;
using System.Net;

namespace Smartproxy_API
{
    class authentication
    {
        static void Main(string[] args)
        {
            string uri = "https://api.smartproxy.com/v1/auth";

            // Your Smartporxy credentials
            string username = "";
            string password = "";

            string encoded = Convert.ToBase64String(System.Text.Encoding.GetEncoding("ISO-8859-1").GetBytes(username + ":" + password));

            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(uri);
            request.AutomaticDecompression = DecompressionMethods.GZip | DecompressionMethods.Deflate;
            request.PreAuthenticate = true;
            request.Headers.Add("Authorization", "Basic " + encoded);
            request.ContentType = "application/json";
            request.Method = "POST";

            using (HttpWebResponse response = (HttpWebResponse)request.GetResponse())
            using (Stream stream = response.GetResponseStream())
            using (StreamReader reader = new StreamReader(stream))
            {
                Console.WriteLine(reader.ReadToEnd());
                Console.ReadKey();
            }
        }
    }
}
